### DISCUSSION TITLE
Multiple git repositories in the same root directory

### DISCUSSION THREAD

**Discussion by Baardi** [gitextensions/gitextensions#11331] [Q&A]
### Feature description

Hi.

Git supports multiple git repositories in the same root directory (https://stackoverflow.com/a/17313342/717372). However this doesn't seem to be supported by Git Extensions. Please correct me if I'm wrong, or if this has been suggested before.

1. Would you consider adding this as a feature?
2. If not prioritized, would pull requests implementing this be welcome?

### Environment

- GitExtensions version:
- GIT version: 
- OS version: 
- .NET version:

**Comment by pmiossec** [comment]
Your request is not really clear.
You provide the link toward a question that don't provide information about what you understand by "Git supports multiple git repositories in the same root directory" and what you expect as feature or support by Git Extensions.

And there are a lot of different cases proposed as answers. Are you referring to a specific answer?

Because for me, if you have 2 repositories in the same folder, you should just open 2 instances of GE, in each repository.
There is no special handling of git (that I know) for this specific case and so I don't think GE should provide a specific handling too...

As a sum up, you should better describe what you mean by "multiple git repositories in the same root directory" (submodules? GE have it.) and/or the features expected.

**Comment by Baardi** [comment]
Submodules cannot be in the same directory as the root of the given repository itself, they have to be added as a sub-directory. I'm referring to something more along the lines with what is described by the .gitone/.gittwo-comment. 

What I'm referring to is to use a different ".git"-direcory (such as .gittwo), where every git-command that is typically performed in the UI replaces git with something along the lines of "git --git-dir=.gittwo".

**Comment by pmiossec** [comment]
Clearer. Thanks (I have updated the message to point toward the good answer)

It seems a hack for me that won't be used by a lot of users and I don't think that's really something we want to support.
But I will let others give their point of view...

**Comment by gerhardol** [comment]
I also see very minor use for the feature and a lot of complexity in implementation and usage.
Separate repos and symbolic links is easier.
